src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenShortCircuitOperatorTests.cs

```diff
@@ -5542,5 +5542,28 @@ public T M()
 M
 ---");
         }
+
+        [WorkItem(74, "https://github.com/dotnet/roslyn/issues/74")]
+        [Fact]
+        public void ConditionalInAsyncTask()
+        {
+            var source = @"
+#pragma warning disable CS1998 // suppress 'no await in async' warning
+using System;
+using System.Threading.Tasks;
+
+class Foo<T>
+{
+    T Method() => default(T); // returns value of unconstrained type parameter type
+    void M1() => this?.Method();
+    async void M2() => this?.Method();
+    async Task M3() => this?.Method();
```

Comment:
Since VB doesn't have "expression lambdas" there is no "corresponding" VB scenario. Please file a specific issue for whatever specific scenario you are concerned about.